spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/pubsub/TracePubSubBeanPostProcessor.java

```diff
@@ -35,10 +33,7 @@
   @Override
   public Object postProcessBeforeInitialization(Object bean, String beanName)
       throws BeansException {
-    if (bean instanceof PublisherFactory) {
-      return new CachingPublisherFactory(
```

Comment:
Would it make sense to just register the customizer here instead of autoconfig? I'm not sure what the tradeoffs are here.

Reply:
That would address the concern of users potentially overriding trace; would also keep all trace-related config in post-processor. I'll try this.

Reply:
As discussed -- tried it, but because only `Publisher.Builder` can have a transformer injected, we'd have to unwrap `DefaultPublisherFactory` and inject a customizer here. It's doable but awkward.